[full-ci] Small fixes

diocas opened the pull request:
Just a bunch of (I hope) fixes:

- I've made the hidden files hidden by default. This is the current behavior in oc10 and I believe should be the same in web;
- Generate sourcemap. We did this to make it easier to debug, let me know if you think this should not be pushed to production (**edit:** made it configurable)
- ~~Clean paths for drawio~~ (opened a diff PR with this fix)

Some more debatable fixes:

~~I've changed the default title of the html page from "owncloud" to "loading" (not translated). Ofc this changes to the configured title once the app loads, but if something fails or takes a long time to load, users get confused.~~ (**edit:** made the title configurable instead)

In the same spirit, I've added information that is displayed before the app loads:  ~~a loading message~~ a spinner that appears if it takes more than 1s) to load resources or an error message if something fails. When we had our performance issues, users were getting blank pages and had no idea what was happening, so a bit of feedback is positive. This is not translated, but the users only see the text on error, so it should be acceptable like this.
Fixes #5961: it doesn't improve the performance, but a least gives feedback to the user instead of a white page

pascalwengerter commented:
@diocas could you run `yarn lint --fix` and commit the changes so we can see what CI says? :)

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20127/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20133/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oCISSharingInternalUsers2</strong> https://drone.owncloud.com/owncloud/web/20139/58/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20175/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20224/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20307/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20311/38/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

diocas commented:
Rebases with latest and did some changes:

- When loading the app takes longer than 1s, instead of showing text, which was in english and was not translated, I show a spinner similar to `ocSpinner`.
- Made the title configurable (since you prob don't want the "loading..." title, as I had before.
- You separated `ActionMenuItem` and changed the display of icon and img to an if/else. But the else is never possible, as an icon of an app is always defined (`icon: appInfo.icon || 'check_box_outline_blank'`), so I inverted it.

@kulmann @pascalwengerter can this be merged? Some of it, at least?

kulmann commented on packages/web-runtime/src/store/apps.js:
```diff
@@ -46,11 +46,13 @@ const mutations = {
       extension: extension.extension,
-      handler: extension.handler
+      handler: extension.handler,
+      canBeDefault: 'canBeDefault' in extension ? extension.canBeDefault : true
```
```suggestion
      canBeDefault: extension.canBeDefault === true
```
has the same effect, right?

diocas replied:
right!

kulmann reviewed (changes requested):
nice, thank you! 🚀 just some small comments, otherwise good to go.

phil-davis commented:
https://drone.owncloud.com/owncloud/web/20581/9/7
`Test Suites: 1 failed, 113 passed, 114 total`

A unit test needs to be fixed.

diocas commented on packages/web-app-files/src/mixins/fileActions.js:
```diff
@@ -62,41 +62,50 @@ export default {
+          }
+        })
+        .sort((first, second) => {
+          // We make the default action based on the order, so put the ones who can be default first
```
Was this a recent fix? In the past, I had non default apps being used when I clicked a file. But maybe the issue was he fact that every app was having `canBeDeFault` as true... I'll modify the comment then.

kulmann replied:
https://github.com/owncloud/web/pull/5970 should fix that. Could you have a look to confirm?

diocas commented on packages/web-app-files/src/mixins/fileActions.js:
```diff
@@ -214,7 +223,8 @@ export default {
         return {
           name: app.name,
-          img: app.icon,
+          icon: app.icon,
```
For us, yes, we added the icons in ODS. But this should be unified in the app provider as well (I will check and ask for changes), as I don't think is intuitive to have them mixed.

diocas commented:
@phil-davis I've changed the test. The test assumed we can delete the icon, but this is not true if you set it from the extension configuration (and since icon always has a default value...). So I removed this part and ensure that, when setting `img` this takes precedence over `icon`.

Also applied @kulmann requests and added a changelog.

kulmann commented:
@wkloucek https://drone.owncloud.com/owncloud/web/20609/9/2 states `msg="execution failed: commit from DRONE_COMMIT_BEFORE not found: object not found"`, could you take a look?

phil-davis commented:
https://drone.owncloud.com/owncloud/web/20609/9/2
That's a strange drone CI fail - nothing to do with the code of this PR!

@diocas we have made a few fixes today for intermittent annoying things in CI pipelines. Please rebase to latest and force-push and you might get more joy! If it dies again the same, then @wkloucek could look or...

kulmann commented:
restarted CI, smoke tests are flaky at the moment, sorry

ownclouders commented:
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/20619/51/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

phil-davis commented:
A smoke-test pipeline failed. I commented in issue https://github.com/owncloud/web/issues/6082#issuecomment-984753681
We are now trying to see why that pipeline is having trouble.
I restarted CI.

ownclouders commented:
Results for <strong>oCISBasic</strong> https://drone.owncloud.com/owncloud/web/20625/51/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

ownclouders commented:
Results for <strong>oC10Trashbin</strong> https://drone.owncloud.com/owncloud/web/20625/40/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

phil-davis commented:
I made PR #6088 on top of this to run full-ci and get the tests passing. If I can achieve that tomorrow morning, then I can push any fixes here to get this passing.

ownclouders commented:
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/20826/16/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108
